Dockerfile

```
@@ -1,30 +1,40 @@
 # syntax = docker/dockerfile:1.3
-ARG ALPINE_VERSION=3.17
+ARG ALPINE_VERSION=3.19
 FROM alpine:${ALPINE_VERSION}
 
 RUN apk add --no-cache git python3 python3-dev py3-pip py3-setuptools build-base
 
-# build wheels in first image
+# build wheels in a build stage
+ENV VIRTUAL_ENV /opt/venv
+ENV PATH ${VIRTUAL_ENV}/bin:${PATH}
+
+RUN python3 -m venv ${VIRTUAL_ENV}
+
 ADD . /tmp/src
 RUN cd /tmp/src && git clean -xfd && git status
 RUN mkdir /tmp/wheelhouse \
  && cd /tmp/wheelhouse \
- && pip3 install wheel \
- && pip3 wheel --no-cache-dir /tmp/src \
+ && pip install wheel \
+ && pip wheel --no-cache-dir /tmp/src \
  && ls -l /tmp/wheelhouse
 
 FROM alpine:${ALPINE_VERSION}
 
 # install python, git, bash, mercurial
 RUN apk add --no-cache git git-lfs python3 py3-pip py3-setuptools bash docker mercurial
 
+ENV VIRTUAL_ENV /opt/venv
+ENV PATH ${VIRTUAL_ENV}/bin:${PATH}
```

```suggestion
ENV VIRTUAL_ENV=/opt/venv
ENV PATH=${VIRTUAL_ENV}/bin:${PATH}
```

Outdated syntax, may be removed, see https://docs.docker.com/reference/dockerfile/#env.